File: web/src/types/api/usuario.ts
import { reserva } from "./reserva";

export interface Usuario {
  id: number | null;
  nome: string;
  sobrenome: string;
  nome_usuario: string;
  senha?: string;
  email: string;
  admin: number;
  motorista: number;
  reservas?: reserva[];
  criado_em: string;
  atualizado_em?: string;
}


export interface paginated_usuarios {
  page: string;
  pages: string;
  items: Usuario[];
}


export interface usuario_co {
  name: string,
  dependants: string[],
  fields: Usuario
}

File: models/usuario.py
from db import db
from datetime import datetime, timezone

class UsuarioModel(db.Model):
  __tablename__ = "usuarios"

  id = db.Column(db.Integer, primary_key=True)
  nome_usuario = db.Column(db.String(25), unique=True, nullable=False)
  nome = db.Column(db.String, nullable=False)
  sobrenome = db.Column(db.String, nullable=True)
  senha = db.Column(db.String, nullable=False)
  email = db.Column(db.String, nullable=False)
  tipo = db.Column(db.Integer, default=0, nullable=False)
  admin = db.Column(db.Boolean, default=0, nullable=False)
  criado_em = db.Column(db.DateTime, default=datetime.now(tz=timezone.utc), nullable=False)
  atualizado_em = db.Column(db.DateTime, nullable=True)

  reservas = db.relationship("ReservaModel", back_populates="usuario", cascade="all, delete")
  viagens = db.relationship("ViagemModel", back_populates="usuario", cascade="all, delete")